include/pando-lib-galois/graphs/dist_local_csr.hpp

```diff
@@ -449,11 +449,34 @@ class DistLocalCSR {
                        numVertices};
   }
 
+  // Gets vertices on this host
+  VertexRange localVertices() {
+    int16_t local_hostID = pando::getCurrentPlace().node.id;
+    int64_t numLocalVertices = getNumEdges(local_hostID);
+    return VertexRange{arrayOfCSRs, lift(arrayOfCSRs[local_hostID], vertexEdgeOffsets.begin),
+                       lift(arrayOfCSRs[local_hostID], vertexEdgeOffsets.end) - 1,
+                       numLocalVertices};
+  }
+
   EdgeRange edges(pando::GlobalPtr<galois::Vertex> vPtr) {
     Vertex v = *vPtr;
     Vertex v1 = *(vPtr + 1);
     return RefSpan<galois::HalfEdge>(v.edgeBegin, v1.edgeBegin - v.edgeBegin);
   }
+
+  EdgeRange edges(pando::GlobalPtr<galois::Vertex> vPtr, uint64_t offset_st, uint64_t window_sz) {
+    Vertex v = *vPtr;
+    Vertex v1 = *(vPtr + 1);
+
+    auto beg = v.edgeBegin + offset_st;
+    if (beg > v1.edgeBegin)
+      return RefSpan<galois::HalfEdge>(v.edgeBegin, 0);
+    if (beg + window_sz <= v1.edgeBegin)
+      return RefSpan<galois::HalfEdge>(beg, window_sz);
+
+    return RefSpan<galois::HalfEdge>(beg, v1.edgeBegin - beg);
+  }
+
```

Comment:
Why do you need to do this? Is it not enough to have doAll infer the locality? Or to just operate on the local csr?

Reply:
- `EdgeRange edges` allows me to chunk the edges for a particular vertex and use that subrange in my `doAll`. This is necessary for chunked TC, as I can execute TC on a subset of outgoing edges each round.
- However, I do see that `localVertices` method is not required as we have a `getLocalCSR.vertices()` I could call instead. I remove `localVertices` in `https://github.com/utcs-scea/pando-mono/pull/31/commits/edb8320623412b48d7747d52b4d43f68638b9e83`

Reply:
okay make sure that works before changing this. But I would prefer that.

Reply:
`getLocalCSR.vertices()` works, but for chunking vertices and edges, I added methods to grab a subset of vertices and edges.